### PR TITLE
Mp/mongoose

### PR DESCRIPTION
**Pull request by milos381**
### Summary
Completely finished Login, SIgnup, provided a mockup page for Dashboard as specified in figma, imported chart, finished oauth to a point where I have to extract the email, which I will do in a new branch.

### Type of Change
Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### Issues
- Link any issues this PR resolves using keywords (resolve, closes, fixed)
#2 , #24, #32 
### Evidence
- Provide evidence of the the changes functioning as expected or describe your tests. If tests are included in the CI pipeline this may be omitted. 

_(delete this line)_ Prior to submitting the PR assign a reviewer from each team to review this PR. 
@donjewey @shalarewicz

### CODE REVIEW

**Comment by milos381** [comment]
Has some merge conflicts  regarding npm packages.  Since I have not done that before, I will leave it up to you guys and you can explain it to me some other time.

**Review by donjewey — APPROVED** [review]
Looks good, we just need to figure out what user information besides email we may need once a user logs in with Github.

File: server/index.ts
Comment (donjewey) [inline server/index.ts:102]: Can you please clean this commenting up before merging?

File: server/index.ts
Comment (donjewey) [inline server/index.ts:32]: It's preferable for our server index.ts file to be cleaner, so before merging can you migrate the auth functionality to an Auth router and controller?

**Review by donjewey — CHANGES_REQUESTED** [review]
I'm requesting a couple of minor changes to clean up the index.ts file

File: server/index.ts
Comment (donjewey) [reply server/index.ts:102]: By this I mean either delete it or move it to a backup file if necessary.

Comment (milos381) [reply server/index.ts:102]: Ok!

File: server/index.ts
Comment (milos381) [reply server/index.ts:32]: Yup!

**Review by donjewey — APPROVED** [review]
Good work!

**Review by shalarewicz — COMMENTED** [review]
Good work! The only comment I have is about how state is used to conditionally display components. Let me know if you want to chat about it.

File: client/App.tsx
Comment (milos381) [reply client/App.tsx:24]: The Dashboard option is just a temporary one, since I needed to design it and implement chart, fetching component, etc... It will be implemented in React Routes with public and private routes.

We can use the logic above for switching between signing up and logging in if you would like.

File: client/App.tsx
Comment (milos381) [reply client/App.tsx:68]: The same reply as above. Dashboard implemented like this is just a temp solution until React Routes are finished.

File: client/components/Loger.tsx
Comment (milos381) [reply client/components/Loger.tsx:10]: Ok.

**Review by shalarewicz — APPROVED** [review]

**Comment by donjewey** [comment]
Please run `npm run lint` and fix the errors in console to allow a merge

**Comment by milos381** [comment]
Merging the base frontend work. All issues regarding this PR were resolved except conditional rendering, but will implement React Routes.

**Pull request merged**
